autokeras/blocks/basic.py

```diff
@@ -154,6 +154,69 @@ def build(self, hp, inputs=None):
         return output_node
 
 
+class AttentionBlock(block_module.Block):
+    """The Attention Block. Luong's Multiplicative Style Attention.
+
+    # Arguments
+        attention_type: String. 'multiplicative' or 'additive'.
+            If left unspecified, it will be tuned automatically.
+
+    """
+
+    def __init__(self,
+                 attention_type: Optional[int] = None,
+                 **kwargs):
+        self.attention_type = attention_type
+        super().__init__(**kwargs)
+
+    def get_config(self):
+        config = super().get_config()
+        config.update({
+            'attention_type': self.attention_type})
+        return config
+
+    def build(self, hp, inputs=None):
+        """
+        # Arguments
+            hp: HyperParameters. The hyperparameters for building the model.
+            inputs: List of the following tensors:
+              * query: Query `Tensor` of shape `[batch_size, Tq, dim]`.
+              * value: Value `Tensor` of shape `[batch_size, Tv, dim]`.
+              * key: Optional key `Tensor` of shape `[batch_size, Tv, dim]`. If not
+                given, will use `value` for both `key` and `value`, which is the
+                most common case.
+
+        # Returns:
+            Attention outputs of shape `[batch_size, Tq, dim]`.
+
+        """
+        inputs = nest.flatten(inputs)
+        attention_type = self.attention_type or hp.Choice('attention_type',
+                                                          ['multiplicative',
+                                                           'additive'],
+                                                          default='additive')
+        attention_layers = {
+            'multiplicative': layers.Attention,
+            'additive': layers.AdditiveAttention
+        }
+        if len(inputs) == 2 or len(inputs) == 3:
+            # Either [query, value] or [query, value, key] provided
+            for in_vec in inputs:
+                if len(in_vec.shape) != 3:
+                    raise ValueError(
+                        'Expect 3 dimensions for each of query, value, (or key) '
+                        'but got {shape}'.format(shape=in_vec.shape)
+                    )
```

Comment:
I expect we have this check in the Keras Attention layers. You may have a try to see the exception the Attention and AdditiveAttention raises. If they have, we can remove this check.

Reply:
If they don't have the exception. We need to check each tensor respectively and give the expected dimensions in details like [batch_size, Tq, dim] in the error message.